lang/rust/avro/src/schema.rs

```diff
@@ -292,6 +317,20 @@ impl Name {
     }
 }
 
+impl Hash for Name {
```

Comment:
Is there a reason to not use derive to implement Hash?

Reply:
According to the [spec](https://avro.apache.org/docs/current/spec.html#names) the `Equality of names is defined on the fullname.`. If we used derivation it will take into account also the optional aliases.